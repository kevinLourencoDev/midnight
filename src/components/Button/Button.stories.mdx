import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';
import { action } from '@storybook/addon-actions';
import Button from './Button';

<Meta title='Button' component={Button} />

# Demo App

## Properties

<Props of={Button} />

## Basic usage

<Preview>
	<Story name='Primary'>
		<Button type="submit" onClick={() => { alert('Button clicked !') }}>Children</Button>
	</Story>
</Preview>
