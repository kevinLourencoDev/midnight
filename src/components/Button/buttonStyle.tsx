import styled from 'styled-components';

const StyledButton = styled.button`
	display: flex;
	align-items: center;
	justify-content: center;
	border-radius: 4px;
	width: 100%;
	height: 40px;
	font-size: 12px;
	letter-spacing: 0.8px;
	text-transform: uppercase;
	cursor: pointer;
	font-weight: bold;

	&:focus {
		outline: none;
	}

	&.primary {
		background-color: #0864cf;
		border: solid 1px #0864cf;
		color: #fff;
		background: linear-gradient(rgba(8, 100, 207, 0.9), #0864cf);

		&:hover {
			background: linear-gradient(rgba(8, 100, 207, 0.8), #0864cf);
		}

		&:active {
		}
	}
`;

export { StyledButton };
