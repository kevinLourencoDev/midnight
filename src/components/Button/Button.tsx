import React, { FC } from 'react';

import { StyledButton } from './buttonStyle';

export interface ButtonProps {
	onClick?: any;
	type?: 'button' | 'submit' | 'reset';
	color?: string;
}

const Button: FC<ButtonProps> = ({
	children,
	onClick,
	type = 'button',
	color = 'primary',
}) => {
	return (
		<StyledButton onClick={onClick} type={type} color={color}>
			{children}
		</StyledButton>
	);
};

export default Button;
