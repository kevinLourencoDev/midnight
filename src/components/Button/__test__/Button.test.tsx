import React from 'react';
import ReactDom from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

import Button from '../Button';

afterEach(cleanup);

describe('Button', () => {
	it('renders without crashing', () => {
		const div: HTMLDivElement = document.createElement('div');
		ReactDom.render(<Button />, div);
		ReactDom.unmountComponentAtNode(div);
	});

	it('matches snapshot', () => {
		const tree = renderer.create(<Button />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Button onClick must be called when button is clicked', () => {
		const mockHandleClick = jest.fn(() => true);

		const button = Enzyme.shallow(<Button onClick={mockHandleClick} />);

		button.find('[type="button"]').simulate('click');
		expect(mockHandleClick.mock.calls.length).toEqual(1);
	});

	it('Button without type should have default type to be button', () => {
		const button = Enzyme.shallow(<Button>test type</Button>);

		expect(button.find('[type="button"]').exists()).toBe(true);
	});

	it('Button with type should have this given type', () => {
		const button = Enzyme.shallow(
			<Button type='submit'>test type submit</Button>
		);

		expect(button.find('[type="submit"]').exists()).toBe(true);
	});

	it('Button without color should have default color to be primary', () => {
		const button = Enzyme.shallow(<Button>test color</Button>);

		expect(button.find('[type="button"]').prop('color')).toEqual('primary');
	});

	it('Button with color should have this given color', () => {
		const button = Enzyme.shallow(<Button color='light'>test color</Button>);

		expect(button.find('[color="light"]').exists()).toBe(true);
	});
});
