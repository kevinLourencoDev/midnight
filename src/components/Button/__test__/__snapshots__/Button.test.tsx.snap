// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<button
  className="sc-bdVaJa kPKiwi"
  color="primary"
  type="button"
/>
`;
